### PR TITLE
SIRo Training Features

### PR DESCRIPTION
**Pull request by ASzot**

### CODE REVIEW

File: habitat-baselines/habitat_baselines/common/construct_vector_env.py [hunk 1]
```diff
@@ -14,6 +14,57 @@
     from omegaconf import DictConfig
 
 
+def grouped_construct_envs(
+    config: "DictConfig",
+    workers_ignore_signals: bool = False,
+    enforce_scenes_greater_eq_environments: bool = False,
+):
```
Comment (xavierpuigf) [inline habitat-baselines/habitat_baselines/common/construct_vector_env.py:21]: Could you add a description of what this does?

File: habitat-baselines/habitat_baselines/config/default_structured_configs.py [hunk 2]
```diff
@@ -378,6 +380,7 @@ class AgentAccessMgrConfig(HabitatBaselinesBaseConfig):
     discrim_reward_weight: float = 1.0
     allow_self_play: bool = False
     self_play_batched: bool = False
+    load_type1_pop_ckpts: Optional[List[str]] = None
```
Comment (xavierpuigf) [inline habitat-baselines/habitat_baselines/config/default_structured_configs.py:383]: Add description of this parameter

File: habitat-lab/habitat/articulated_agent_controllers/humanoid_rearrange_controller.py [hunk 3]
```diff
@@ -152,8 +152,9 @@ def calculate_walk_pose(
         :param distance_multiplier: allows to create walk motion while not translating, good for turning
         """
         deg_per_rads = 180.0 / np.pi
+
         forward_V = target_position
-        if forward_V.length() < EPS:
+        if forward_V.length() < EPS or np.isnan(target_position).any():
```
Comment (xavierpuigf) [inline habitat-lab/habitat/articulated_agent_controllers/humanoid_rearrange_controller.py:157]: Should we instead raise an exception if target_position is NaN?

File: habitat-lab/habitat/tasks/rearrange/rearrange_sim.py [hunk 4]
```diff
@@ -253,26 +254,43 @@ def reset(self):
...
 
-        with read_write(config):
-            config["scene"] = ep_info.scene_id
-
         self.ep_info = ep_info
         new_scene = self.prev_scene_id != ep_info.scene_id
         if new_scene:
             self._prev_obj_names = None
 
+        if self._force_soft_reset:
```
Comment (xavierpuigf) [inline habitat-lab/habitat/tasks/rearrange/rearrange_sim.py:264]: Are these two objects also the target objects to rearrange?

File: habitat-baselines/habitat_baselines/rl/multi_agent/sensors.py [hunk 5]
```diff
@@ -176,6 +176,6 @@ def get_observation(self, *args, **kwargs):
                 entities = self._action_map[use_name]
                 set_idx += entities.index(param_value)
                 self._dat[self._cur_write_idx, set_idx] = 1.0
-            self._cur_write_idx += 1
```
[the remark this answers is not in the document]
Comment (ASzot) [reply habitat-baselines/habitat_baselines/rl/multi_agent/sensors.py:179]: It wasn't a bug necessarily, it just made the discriminator job much harder. This line kept inserting no op actions when the agent was not taking any actions. Removing this line only inserts actions for HL policy decisions.

File: habitat-lab/habitat/tasks/rearrange/rearrange_sim.py (hunk 4 above)
Comment (ASzot) [reply habitat-lab/habitat/tasks/rearrange/rearrange_sim.py:264]: Yes, I added a comment to address these as well.

File: habitat-lab/habitat/articulated_agent_controllers/humanoid_rearrange_controller.py (hunk 3 above)
Comment (ASzot) [reply habitat-lab/habitat/articulated_agent_controllers/humanoid_rearrange_controller.py:157]: Unfortunately this frequently happens in my code. So raising an exception would crash the run. We should figure out what the long term fix is.

File: habitat-baselines/habitat_baselines/config/habitat_baselines/rl/policy/hierarchical_policy/defined_skills/oracle_skills_ma.yaml [hunk 6]
```diff
@@ -62,3 +62,20 @@ reset_arm:
   max_skill_steps: 50
   reset_joint_state: [-4.50e-01, -1.07e00, 9.95e-02, 9.38e-01, -7.88e-04, 1.57e00, 4.62e-03]
   force_end_on_timeout: False
+
+
+turn_left:
+  skill_name: "TurnSkillPolicy"
+  force_end_on_timeout: False
+  max_skill_steps: 5
```
[the remark this answers is not in the document]
Comment (ASzot) [reply habitat-baselines/habitat_baselines/config/habitat_baselines/rl/policy/hierarchical_policy/defined_skills/oracle_skills_ma.yaml:70]: Good point, I removed `turn_steps`.

**Review by xavierpuigf — APPROVED** [review]
Thanks for addressing the changes! LGTM

**Pull request merged**